faf-commons-data/src/main/java/com/faforever/commons/replay/body/ReplayBodyParser.java

```diff
@@ -320,6 +262,12 @@ private static Event parseToken(Token token) throws IOException {
 
         case null -> new Event.Unprocessed(token, "Unknown");
       };
+
+      if(stream.available() > 0) {
+        throw new EOFException();
+      }
```

Comment:
This a little confusing use of exception. typically EOFException means we hit the end of the file when we are expecting more data. Here we throw an exception if there is data left over.

I think probably just an illegal state exception with appropriate message is best

Reply:
I agree it is semantically better, I was searching for a proper exception when writing the code. Processed it in:

- https://github.com/FAForever/faf-java-commons/pull/143/commits/4348c647a68b0f0d2bc59b3aac87f352297d5f20